src/main/kotlin/com/jfrog/conan/clionplugin/conan/Conan.kt

```diff
@@ -1,50 +1,83 @@
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
 import com.jfrog.conan.clionplugin.models.PersistentStorageKeys
 import java.io.File
 
 class Conan(val project: Project) {
 
-    data class RunOutout (
-        val exitCode: Int,
-        val stdout: String,
-        val stderr: String
+    data class RunOutput(
+            val exitCode: Int,
+            val stdout: String,
+            val stderr: String
     )
 
-    private fun run(args: List<String>): RunOutout {
-        val conanExecutable: String = this.project.service<PropertiesComponent>().getValue(PersistentStorageKeys.CONAN_EXECUTABLE, "conan")
-        val command = listOf(conanExecutable) + args
-        thisLogger().info("Running command: $command")
+    private fun runInBackground(args: List<String>, taskTitle: String, onSuccess: (RunOutput) -> Unit) {
+        val task = object : Task.Backgroundable(project, taskTitle, true) {
+            override fun run(indicator: ProgressIndicator) {
+                indicator.isIndeterminate = true
+                val conanExecutable: String = project.service<PropertiesComponent>().getValue(
+                        PersistentStorageKeys.CONAN_EXECUTABLE,
+                        "conan"
+                )
+                val command = listOf(conanExecutable) + args
+                thisLogger().info("Running command: $command")
 
-        val process = ProcessBuilder(command)
-                .directory(File(project.basePath!!))
-                .redirectOutput(ProcessBuilder.Redirect.PIPE)
-                .redirectError(ProcessBuilder.Redirect.PIPE)
-                .start()
+                val commandLine = GeneralCommandLine(command)
+                        .withWorkDirectory(File(project.basePath!!))
 
-        val stdout = process.inputStream.bufferedReader().use { it.readText() }
-        val stderr = process.errorStream.bufferedReader().use { it.readText() }
+                val processHandler = CapturingProcessHandler(commandLine)
 
-        val exitCode = process.waitFor()
+                var exitCode: Int = Int.MIN_VALUE
+                var stdout = ""
+                var stderr = ""
 
-        thisLogger().info("Command exited with status $exitCode")
-        thisLogger().info("Command stdout: $stdout")
-        thisLogger().warn("Command stderr: $stderr")
+                processHandler.addProcessListener(object : CapturingProcessAdapter() {
+                    override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                        if (outputType == ProcessOutputType.STDOUT) {
+                            stdout += event.text
+                        } else if (outputType == ProcessOutputType.STDERR) {
+                            stderr += event.text
+                        }
+                        var line = event.text.trim()
+                        indicator.text2 = line
+                        if (line.startsWith("========")) {
```

Comment:
this here is to show the main sections of the conan commands in the output, the conan output is tipically something like:

```
======== Computing dependency graph ========
Graph root
    cli
Requirements
    jander/0.1.0#1013230aab985333c7ca793066ce84aa - Cache

======== Computing necessary packages ========
jander/0.1.0: Forced build from source
Requirements
    jander/0.1.0#1013230aab985333c7ca793066ce84aa:3a30e794f52cc541a8eef1d48bd278ebdcc60c18 - Build
````
and the sections start with `"========"`